src/utils/ICS24Host.sol

```diff
@@ -152,4 +153,18 @@ library ICS24Host {
     function packetAcknowledgementCommitmentBytes32(bytes memory ack) internal pure returns (bytes32) {
         return sha256(ack);
     }
+
+    /// @notice Create a prefixed path
+    /// @dev The path is appended to the last element of the prefix
+    /// @param merklePrefix The prefix
+    /// @param path The path to append
+    /// @return The prefixed path
+    function prefixedPath(bytes[] memory merklePrefix, bytes memory path) internal pure returns (bytes[] memory) {
```

Comment:
I found the expected API to be a little confusing. Because we now expect the counterparty prefix to already have an empty element. I don't like the UX of that too much. There is a slight bump of gas (a couple of k) to do this:

```solidity
function prefixedPath(bytes[] memory merklePrefix, bytes memory path) internal pure returns (bytes[] memory) {
    if (merklePrefix.length == 0) {
        revert IICS24HostErrors.InvalidMerklePrefix(merklePrefix);
    }

    bytes[] memory prefixedPath = new bytes[](merklePrefix.length+1);
    for (uint256 i = 0; i < merklePrefix.length; i++) {
        prefixedPath[i] = merklePrefix[i];
    }
    prefixedPath[merklePrefix.length] = path;

    return prefixedPath;
}
```

Or, if we prefer to save the gas, and keep the empty path ready, we could add an empty element to the counterparty prefix on addClient. That way the users of the this doesn't need to know about it and we only take that hit on addClient.

Reply:
But this is the way it is done in ibc-go and ics23 I think.

Reply:
I was not able to find that. Could you check if that is the case?

Reply:
Ah, I think you're right that this is not how ibc-go does it at the moment. But the current implementation in ibc-go is actually incorrect and not generalized properly. We can create an issue in ibc-go, or I can let Aditya know.